library/aes.c

```diff
@@ -525,12 +533,16 @@ void mbedtls_aes_free( mbedtls_aes_context *ctx )
 #if defined(MBEDTLS_CIPHER_MODE_XTS)
 void mbedtls_aes_xts_init( mbedtls_aes_xts_context *ctx )
 {
+    MBEDTLS_AES_VALIDATE( ctx != NULL );
+
     mbedtls_aes_init( &ctx->crypt );
     mbedtls_aes_init( &ctx->tweak );
 }
 
 void mbedtls_aes_xts_free( mbedtls_aes_xts_context *ctx )
 {
+    MBEDTLS_AES_VALIDATE( ctx != NULL );
```

Comment:
Why the asymmetry with mbedtls_aes_free()? Apparently  mbedtls_aes_free() can handle a NULL ctx (it just does nothing), but mbedtls_aes_xts_free() cannot? I suggest making both of them behave the same.